TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/NettyPacketToHttpConsumer.java

```diff
@@ -45,112 +85,120 @@ public NettyPacketToHttpConsumer(NioEventLoopGroup eventLoopGroup, URI serverUri
         String host = serverUri.getHost();
         int port = serverUri.getPort();
         log.debug("Active - setting up backend connection to " + host + ":" + port);
-        outboundChannelFuture = b.connect(host, port);
+        var outboundChannelFuture = b.connect(host, port);
         //outboundChannel = outboundChannelFuture.channel();
         responseWatchHandler = new BacksideHttpWatcherHandler(responseBuilder);
+        fullyInitializedChannelFuture =
+                new StringTrackableCompletableFuture<>(new CompletableFuture<>(), () -> "sslHandshakeFuture");
 
         outboundChannelFuture.addListener((ChannelFutureListener) future -> {
             if (future.isSuccess()) {
                 // connection complete start to read first data
                 log.debug(diagnosticLabel + "Done setting up backend channel & it was successful");
                 var pipeline = future.channel().pipeline();
+                pipeline.addFirst(new LoggingHandler("PRE_EVERYTHING", LogLevel.TRACE));
                 if (sslContext != null) {
-                    SSLEngine sslEngine = sslContext.newEngine(future.channel().alloc());
+                    sslEngine = sslContext.newEngine(future.channel().alloc());
                     sslEngine.setUseClientMode(true);
-                    pipeline.addFirst("ssl", new SslHandler(sslEngine));
+                    var sslHandler = new SslHandler(sslEngine);
+                    pipeline.addLast("ssl", sslHandler);
+                    pipeline.addLast(new LoggingHandler("POST_SSL", LogLevel.TRACE));
+                    pipeline.addLast(new SslErrorHandler(e-> terminateChannel(pipeline)));
+                    pipeline.addLast(new LoggingHandler("POST_ERROR_CATCHER", LogLevel.TRACE));
+                    sslHandler.handshakeFuture().addListener(f -> {
+                        if (f.isSuccess()) {
+                            addHttpTransactionHandlersToPipeline(pipeline);
+                            fullyInitializedChannelFuture.future.complete(future.channel());
+                        } else {
+                            fullyInitializedChannelFuture.future.completeExceptionally(future.cause());
+                        }
```

Comment:
Is this essentially saying that if my ssl handshake fails I will not add further handlers to the pipeline? Are there existing handlers that this message could still go to?

Reply:
The only other ones in the pipeline at this point are diagnostic ones.  addHttpTransactionHandlersToPipeline adds the other ones, but only after a successful handshake